server/internal/mcp/impl.go

```diff
@@ -708,27 +753,58 @@ func parseMcpSessionID(headers http.Header) string {
 	return session
 }
 
-func (s *Service) authenticateToken(ctx context.Context, token string) (context.Context, error) {
-	// This just follows Goa's implementation of checking multiple key scopes as a union
-	// Adding both scopes to the same RequiredScopes [] implies both scopes being required
+func (s *Service) authenticateToken(ctx context.Context, token string, toolsetID uuid.UUID) (context.Context, error) {
```

Comment:
nit maybe we should pass isOauthCapable into here. So we don't waste our time with that `s.oauthService.ValidateAccessToken` if we already know the server doesn't have oauth

Reply:
You're absolutely right!